ProcessLib/BoundaryCondition/BoundaryCondition.cpp

```diff
@@ -10,53 +10,42 @@
 #include "BoundaryCondition.h"
 #include "MeshGeoToolsLib/BoundaryElementsSearcher.h"
 #include "MeshGeoToolsLib/MeshNodeSearcher.h"
+#include "MeshGeoToolsLib/SearchLength.h"
 #include "BoundaryConditionConfig.h"
 #include "DirichletBoundaryCondition.h"
 #include "NeumannBoundaryCondition.h"
 #include "RobinBoundaryCondition.h"
 
 namespace ProcessLib
 {
-
-std::unique_ptr<BoundaryCondition> BoundaryConditionBuilder::createBoundaryCondition(
+std::unique_ptr<BoundaryCondition>
+BoundaryConditionBuilder::createBoundaryCondition(
     const BoundaryConditionConfig& config,
     const NumLib::LocalToGlobalIndexMap& dof_table, const MeshLib::Mesh& mesh,
     const int variable_id, const unsigned integration_order,
     const unsigned shapefunction_order,
     const std::vector<std::unique_ptr<ProcessLib::ParameterBase>>& parameters)
 {
-    MeshGeoToolsLib::SearchLength search_length_algorithm;
```

Comment:
The `SearchLength` object can only be created after parsing the boundary condition. So the creation is moved there.